### PR TITLE
Observ rounding and composite tests

### PR DESCRIPTION
**Pull request by daco101**
This PR limits observ expected values to 8 decimal places, and also will only compare (and display) expected and actual observ values up to 8 decimal places. This should make it more in line with the cql spec, but maybe only the input limiting would be needed if cql-exec were fully compliant.
This should fix an issue where it is impossible to correctly match actual and expected observs.

This PR also introduces some improved composite measure testing, as well as coverage testing.

Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1812
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] NA If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced.
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Test fixtures updated and documented as necessary ( see [internal wiki](https://gitlab.mitre.org/bonnie/internal-documentation/wikis/testing#test-fixtures) )
- [x] Code coverage has not gone down and all code touched or added is covered. 
     * In rare situations, this may not be possible or applicable to a PR. In those situations:
         1. Note why this could not be done or is not applicable here: 
         2. Add TODOs in the code noting that it requires a test
         3. Add a JIRA task to add the test and link it here: 

Branch | Back End Coverage | Front End Coverage
-- | -- | --
bonnie-v3.1 | 2226 / 2487 LOC (89.51%) covered | Statements   : 60.25% ( 7445/12356 )<br/>Branches     : 45.23% ( 2091/4623 )<br/>Functions    : 50.7% ( 1235/2436 )<br/>Lines        : 58.19% ( 6692/11501 )
observ_rounding_and_composite_tests | 2226 / 2487 LOC (89.51%) covered| Statements   : 60.45% ( 7474/12364 )<br/>Branches     : 45.86% ( 2121/4625 )<br/>Functions    : 50.8% ( 1239/2439 )<br/>Lines        : 58.4% ( 6721/11509 )

- [x] Automated regression test(s) pass -losborne

If JIRA tests were used to supplement or replace automated tests:
- [x] NA JIRA test links:
- [x] NA Justification for using JIRA tests:
- [x] NA JIRA tests have been added to sprint

**Reviewer 1:**

Name: @losborne
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name: @mayerm94 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

File: app/assets/javascripts/views/patient_builder/expected_values.js.coffee [hunk 1]
```diff
@@ -62,7 +62,7 @@ class Thorax.Views.ExpectedValueView extends Thorax.Views.BuilderChildView
           if attr[pc]
             if pc == 'OBSERV'
               attr[pc] = [].concat(attr[pc])
-              attr[pc] = (@roundToCQLPrecision(parseFloat(o)) for o in attr[pc])
+              attr[pc] = (Thorax.Models.ExpectedValue.floorToCQLPrecision(parseFloat(o)) for o in attr[pc])
             else
               attr[pc] = parseFloat(attr[pc])
```
Comment (lukoktonos) [inline app/assets/javascripts/views/patient_builder/expected_values.js.coffee:67]: Should we also limit decimal places for this `parseFloat`?

Comment (daco101) [reply app/assets/javascripts/views/patient_builder/expected_values.js.coffee:67]: Any ideas when an expected value is a number but not an observ?

Comment (lukoktonos) [reply app/assets/javascripts/views/patient_builder/expected_values.js.coffee:67]: No, perhaps this `parseFloat` should really just be a `+` since they are intended to be integer values.  I suppose just leave it for now since it's beyond the scope of this PR.

**Review by lukoktonos — APPROVED** [review]

**Review by mayerm94 — APPROVED** [review]
